lib/cli/routes.js

```diff
@@ -0,0 +1,42 @@
+import nopt from 'nopt';
+import {tablify} from '../util.js';
+
+export default function routesCommand (app, args) {
+  const parsed = nopt({verbose: Boolean}, {v: '--verbose'}, args, 1);
+  const table = [];
+  app.router.children.map(route => _walk(route, 0, table, parsed.verbose));
+  process.stdout.write(tablify(table));
+}
+
+routesCommand.description = 'Show available routes';
+routesCommand.usage = `Usage: APPLICATION routes [OPTIONS]
+
+  ./myapp.js routes
+  ./myapp.js routes -v
+
+Options:
+  -h, --help      Show this summary of available options
+  -v, --verbose   Print additional details about routes
+`;
+
+function _walk (route, depth, rows, verbose) {
+  const prefix = ' '.repeat(depth * 2);
+  const row = [prefix + (route.pattern.unparsed || '/') + (depth ? '+' : '')];
+  rows.push(row);
+
+  // Methods
+  row.push(route.methods.length === 0 ? '*' : route.methods.map(method => method.toUpperCase()).join(','));
+
+  // Name
+  row.push(typeof route.customName !== 'undefined' ? `"${route.customName}"` : route.defaultName);
```

Comment:
Why is that better than `route.customName !== undefined`?

Reply:
in some js universes `undefined` is a variable which can have a value

Reply:
🙄